R/lat_bins.R

```diff
@@ -28,80 +39,84 @@
   }
 
   if (is.logical(fit) == FALSE) {
-    stop("fit should be logical (TRUE/FALSE)")
+    stop("`fit` should be logical (TRUE/FALSE)")
   }
 
-  if (sum(assign > 90) != 0 | sum(assign < -90) != 0) {
-    stop("Latitude should be more than -90 and less than 90")
+  if (any(assign > 90 | assign < -90)) {
+    stop("Latitudes (`assign`) should be more than -90 and less than 90")
   }
 
   if (is.logical(plot) == FALSE) {
-    stop("plot should be logical (TRUE/FALSE)")
+    stop("`plot` should be logical (TRUE/FALSE)")
   }
 
   #divide latitudinal range by size of bins
-  bins <- 180/size
+  bins <- 180 / size
   #if fit is set true, generate equal size bins to fit range
-  if(fit == TRUE){
-    if(is.integer(bins) == FALSE){
-      int <- 180/seq(from = 1, to = 90, by = 1)
-      int <- which(int%%1==0)
+  if (fit == TRUE) {
+    if (is.integer(bins) == FALSE) {
+      int <- 180 / seq(from = 1, to = 90, by = 1)
+      int <- which(int %% 1 == 0)
       size <- int[which.min(abs(int - size))]
-      bins <- 180/size
+      bins <- 180 / size
     }
   }
   #generate latitudinal bins
   df <- seq(from = -90, to = 90, by = size)
   min <- df[1:bins]
-  max <- df[1:bins]+size
-  mid <- (max+min)/2
+  max <- df[1:bins] + size
+  mid <- (max + min) / 2
   bin <- 1:bins
   df <- cbind(max, mid, min)
-  df <- df[order(-max),]
+  df <- df[order(-max), ]
   df <- cbind.data.frame(bin, df)
   #plot latitudinal bins
-  if(plot == TRUE){
-    plot(1, type = "n", xlim = c(-180, 180), ylim = c(min(df$min), max(df$max)), xlab = "Longitude (\u00B0)", ylab = "Latitude (\u00B0)")
-    cols <- rep(c("#2166ac", "#b2182b"), nrow(df))
-    for(i in 1:nrow(df)){
+  if (plot == TRUE) {
+    plot(1, type = "n", xlim = c(-180, 180),
+         ylim = c(min(df$min), max(df$max)),
+         xlab = "Longitude (\u00B0)", ylab = "Latitude (\u00B0)")
+    cols <- rep(c("#2ca25f", "#ccece6"), nrow(df))
+    for (i in seq_len(nrow(df))){
       polygon(x = c(-180, -180, 180, 180),
               y = c(df$min[i], df$max[i], df$max[i], df$min[i]),
               col = cols[i],
               border = "black")
     }
-    if(fit == TRUE){
+    if (fit == TRUE) {
       title(paste0("Bin size set to ", size))
     }
   }
 
-  if(fit == TRUE){
-    message(paste0("Bin size set to ",size," degrees to fit latitudinal range."))
+  if (fit == TRUE) {
+    message(paste0(
+      "Bin size set to ", size, " degrees to fit latitudinal range."))
   }
 
-  if(!is.null(assign)){
-    if(is.numeric(assign)){
-      if(any(assign > 90 | assign < -90)){
-        stop("One or more latitudes is more than or less than 90/-90")
-     }
+  if (!is.null(assign)) {
+    if (is.numeric(assign)) {
     tmp <- assign
-      for(i in 1:nrow(df)){
+      for (i in seq_len(nrow(df))){
         assign[which(tmp <= df$max[i] & tmp >= df$min[i])] <- df$mid[i]
+        names(assign)[which(tmp <= df$max[i] &
+                              tmp >= df$min[i])] <- df$bin[i]
```

Comment:
This is a really good point. I've updated the argument description for assign. Currently, the occurrences will always be allocated to the bin with the largest bin number. I think this is OK.. but yes, there could be instances where you might want to place in both bins. However, I would favour adding this as an issue for now? This would require a bit of rejigging of the output otherwise and it would become a bit more challenging for the user to add the bins directly to their data.

Reply:
Agreed, especially as I think it would only be needed in fairly specialist cases.

Reply:
Perfect! I'll log an issue!